ch_tools/chadmin/cli/part_group.py

```diff
@@ -559,3 +560,98 @@ def move_parts_command(
                 logging.warning("{!r}\n", e)
             else:
                 raise
+
+
+@part_group.command("remove-detached-part-prefix")
+@option_group(
+    "Part selection options",
+    option(
+        "-a",
+        "--all",
+        "_all",
+        is_flag=True,
+        help="Filter in all data parts.",
+    ),
+    option(
+        "-d",
+        "--database",
+        help="Filter in data parts to attach by the specified database.",
+    ),
+    option(
+        "-t", "--table", help="Filter in data parts to attach by the specified table."
+    ),
+    option(
+        "--id",
+        "--partition",
+        "partition_id",
+        help="Filter in data parts to attach by the specified partition.",
+    ),
+    option(
+        "--name",
+        "--part",
+        "part_name",
+        help="Filter in data parts to attach by the specified data part name.",
+    ),
+    option(
+        "--disk",
+        "disk_name",
+        help="Filter in data parts to attach by the specified disk.",
+    ),
+    option(
+        "-l",
+        "--limit",
+        type=int,
+        help="Limit the max number of data parts to attach.",
+    ),
+    option(
+        "--use-part-list-from-json",
+        default=None,
+        type=str,
+        help="Use list of parts from the file. Example 'SELECT database, table, name ... FORMAT JSON' > file && chadmin part attach --use-part-list-from-json <file>.",
+    ),
+    constraint=If(
+        IsSet("use_part_list_from_json"),
+        then=RequireExactly(1),
+        else_=RequireAtLeast(1),
+    ),
+)
+@option("-k", "--keep-going", is_flag=True, help="Do not stop on the first error.")
+@option(
+    "-n",
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    help="Enable dry run mode and do not perform any modifying actions.",
+)
+@option(
+    "--reason",
+    default="broken-on-start",
+    type=str,
+    help="Part reason of detach.",
+)
+@pass_context
+def remove_detached_part_prefix_command(
+    ctx: Context,
+    keep_going: bool,
+    dry_run: bool,
```

Comment:
Having a separate command can be helpful, but it makes it impossible to do attach or drop only for parts with specific prefix, I think it can be called in some other commands from the group if detach reason is specified